keyring/backends/macOS/__init__.py

```diff
@@ -32,36 +33,54 @@ def priority(cls):
             raise RuntimeError("Security API unavailable")
         return 5
 
-    def set_password(self, service, username, password):
+    def set_password(self, service, username, password, keychain=""):
         if username is None:
             username = ''
 
+        if keychain != "":
+            if not exists(keychain):
+                raise KeyringLocked("Can't open specified keychain")
+        elif os.environ.get('KEYCHAIN_PATH') != "":
+            keychain = os.environ.get('KEYCHAIN_PATH')
+
```

Comment:
There's a whole block of logic here that adds to the complexity of the set_password method. Better than to have it inline would be to have it as a method or function or property that describes its purpose.